### PR TITLE
userlayerdata: return simpleFeatureCollection from mybatis

### PR DESCRIPTION
**Pull request by DenverCoder544**
Simplify implementation and just return simplefeaturecollection from the results of db. Got rid of many unnecessary transformations (format, srs).

### CODE REVIEW

**Comment by jampukka** [comment]
Could we get by with single FeatureType with nullable Geometry.class?

File: service-userlayer/src/main/java/org/oskari/map/userlayer/service/UserLayerDbServiceMybatisImpl.java [hunk 1]
```diff
@@ -266,74 +277,107 @@ private UserLayerMapper getMapper(SqlSession session) {
...
-        } catch(JSONException ex) {
-            log.warn("Failed to convert UserLayerData to GeoJSONFeature");
+    private SimpleFeature toSimpleFeature(UserLayerData feature) {
+
+        Geometry geom = parseWKT(feature.getWkt());
+        SimpleFeatureTypeBuilder featureTypeBuilder = getFeatureTypeBuilder(geom);
+
+        SimpleFeatureType simpleFeatureType = featureTypeBuilder.buildFeatureType();
+        SimpleFeatureBuilder featureBuilder = new SimpleFeatureBuilder(simpleFeatureType);
+        featureBuilder.set(GEOM_ATTRIBUTE, parseWKT(feature.getWkt()));
```
Comment (jampukka) [inline service-userlayer/src/main/java/org/oskari/map/userlayer/service/UserLayerDbServiceMybatisImpl.java:320]: Don't parse WKT again, use the already parsed `geom` instead.

Comment (DenverCoder544) [reply service-userlayer/src/main/java/org/oskari/map/userlayer/service/UserLayerDbServiceMybatisImpl.java:320]: You're right about these. Unfortunately it seems that a generic featuretype with Geometry.class does not work. It results in an exception when trying to create the actual feature.

Comment (jampukka) [reply service-userlayer/src/main/java/org/oskari/map/userlayer/service/UserLayerDbServiceMybatisImpl.java]: Thanks for trying!

**Pull request merged**